Issue with called.withArg or Args after 0.4.1

Using 0.4.0 the test sample below passes, but >=0.4.1 fails. Some issue with the upgrade of lodash to 4.5.1.

```js
it('booooom', function() {
    var obj = deride.stub(['send']);
    obj.send(['a', 'b'], function() {});
    obj.expect.send.called.withArgs(['a', 'b']);
});

```
produces:

```shell
  Expectations
    1) booooom


  0 passing (14ms)
  1 failing

  1) Expectations booooom:

      AssertionError: Expected send to be called with: a,b
      + expected - actual

      -false
      +true

      at assertArgsWithEvaluator (lib/deride.js:118:9)
      at Object.withArgs (lib/deride.js:66:9)
      at Context.<anonymous> (test/test-deride.js:124:32)
```

--- a/lib/deride.js
+++ b/lib/deride.js
@@ -39,8 +39,12 @@ function Expectations(obj, method) {
     var calledWithArgs = {};
 
     function checkArg(expected, values) {
-        if (_.isObject(expected)) {
+        if (_.isPlainObject(expected)) {
             return _.some(_.filter(values, expected));
+        }
+        if (_.isArray(expected)) {
+            var jsonExpected = JSON.stringify(expected);
+            return _.some(_.filter(values, function (v) { return JSON.stringify(v) === jsonExpected; }));
         }
         return _.includes(values, expected);
     }

--- a/lib/deride.js
+++ b/lib/deride.js
@@ -39,12 +39,12 @@ function Expectations(obj, method) {
     var calledWithArgs = {};
 
     function checkArg(expected, values) {
-        if (_.isPlainObject(expected)) {
-            return _.some(_.filter(values, expected));
-        }
         if (_.isArray(expected)) {
             var jsonExpected = JSON.stringify(expected);
             return _.some(_.filter(values, function (v) { return JSON.stringify(v) === jsonExpected; }));
+        }
+        if (_.isObject(expected)) {
+            return _.some(_.filter(values, expected));
         }
         return _.includes(values, expected);
     }

--- a/package.json
+++ b/package.json
@@ -37,7 +37,7 @@
     "grunt-notify": "~0.4.1",
     "istanbul": "^0.3.17",
     "mocha": "^2.2.5",
-    "should": "^7.0.1"
+    "should": "^8.3.0"
   },
   "keywords": [
     "test",

--- a/test/test-deride.js
+++ b/test/test-deride.js
@@ -118,6 +118,40 @@ describe('Expectations', function() {
         done();
     });
 
+    describe('withArg called with an array', function () {
+        var obj;
+        beforeEach(function () {
+            obj = deride.stub(['send']);
+            obj.setup.send.toReturn('talula');
+        });
+
+        _.forEach([{
+            name: 'string', input: 'talula', expectPass: false
+        }, {
+            name: 'non match array', input: ['d', 'e'], expectPass: false
+        }, {
+            name: 'partial match array', input: ['a', 'd'], expectPass: false
+        }, {
+            name: 'match but wrong order', input: ['b', 'a'], expectPass: false
+        }, {
+            name: 'match', input: ['a', 'b'], expectPass: true
+        }], function (test) {
+            if (test.expectPass) {
+                it('object called with ' + test.name + ' should pass', function() {
+                    obj.send(test.input);
+                    obj.expect.send.called.withArg(['a', 'b']);
+                });
+            } else {
+                it('object called with ' + test.name + ' should fail', function() {
+                    obj.send(test.input);
+                    assert.throws(function() {
+                        obj.expect.send.called.withArg(['a', 'b']);
+                    }, Error);
+                });
+            }
+        });
+    });
+
     it('handles comparison of withArgs when an argument is a function', function(done) {
         var obj = deride.stub(['send']);
         obj.send({
@@ -542,22 +576,12 @@ _.forEach(tests, function(test) {
             bob.setup.greet.when('norman').toRejectWith(new Error('foobar'));
         });
 
-        it('enables resolving a promise', function(done) {
-            bob.greet('alice').then(function(result) {
-                assert.equal(result, 'foobar');
-                done();
-            });
+        it('enables resolving a promise', function() {
+            bob.greet('alice').should.be.fulfilledWith('foobar');
         });
 
-        it('enables rejecting a promise', function(done) {
-            bob.greet('norman')
-                .then(function() {
-                    done('should not have resolved');
-                })
-                .catch(function(result) {
-                    assert.equal(result.message, 'foobar');
-                    done();
-                });
+        it('enables rejecting a promise', function() {
+            bob.greet('norman').should.be.rejectedWith('foobar');
         });
     });
 

--- a/test/test-deride.js
+++ b/test/test-deride.js
@@ -81,27 +81,31 @@ describe('utils', function() {
 
 });
 
-describe('Expectations', function() {
-    it('does not invoke original method when override method body', function() {
-        var obj = deride.stub(['send']);
-        obj.setup.send.toThrow('bang');
+describe.only('Expectations', function() {
+    var bob;
+    beforeEach(function () {
+        bob = deride.stub(['greet']);
+        bob.setup.greet.toReturn('talula');
+    });
 
-        obj = deride.wrap(obj);
-        obj.setup.send.toDoThis(function() {
+    it('does not invoke original method when override method body', function() {
+        bob.setup.greet.toThrow('bang');
+
+        bob = deride.wrap(bob);
+        bob.setup.greet.toDoThis(function() {
             return 'hello';
         });
-        var result = obj.send();
+        var result = bob.greet();
         assert.equal(result, 'hello');
     });
 
     it('ignores the order of an object properties when comparing equality', function(done) {
-        var obj = deride.stub(['send']);
-        obj.send({
+        bob.greet({
             c: 3,
             b: 2,
             a: 1
         });
-        obj.expect.send.called.withArgs({
+        bob.expect.greet.called.withArgs({
             a: 1,
             b: 2,
             c: 3
@@ -110,21 +114,14 @@ describe('Expectations', function() {
     });
 
     it('throws exception when object not called withArgs', function(done) {
-        var obj = deride.stub(['send']);
-        obj.send('1', '2', '3');
+        bob.greet('1', '2', '3');
         assert.throws(function() {
-            obj.expect.send.called.withArgs('4');
+            bob.expect.greet.called.withArgs('4');
         }, Error);
         done();
     });
 
     describe('withArg called with an array', function () {
-        var obj;
-        beforeEach(function () {
-            obj = deride.stub(['send']);
-            obj.setup.send.toReturn('talula');
-        });
-
         _.forEach([{
             name: 'string', input: 'talula', expectPass: false
         }, {
@@ -138,33 +135,35 @@ describe('Expectations', function() {
         }], function (test) {
             if (test.expectPass) {
                 it('object called with ' + test.name + ' should pass', function() {
-                    obj.send(test.input);
-                    obj.expect.send.called.withArg(['a', 'b']);
+                    bob.greet(test.input);
+                    bob.expect.greet.called.withArg(['a', 'b']);
                 });
             } else {
                 it('object called with ' + test.name + ' should fail', function() {
-                    obj.send(test.input);
+                    bob.greet(test.input);
                     assert.throws(function() {
-                        obj.expect.send.called.withArg(['a', 'b']);
+                        bob.expect.greet.called.withArg(['a', 'b']);
                     }, Error);
                 });
             }
         });
     });
 
-    it('handles comparison of withArgs when an argument is a function', function(done) {
-        var obj = deride.stub(['send']);
-        obj.send({
+    it('handles withArg called with object', function () {
+        bob.greet(new Error('booom'));
+        bob.expect.greet.called.withArgs(new Error('booom'));
+    });
+
+    it('handles comparison of withArgs when an argument is a function', function() {
+        bob.greet({
             a: 1
         }, function() {});
-        obj.expect.send.called.withArgs({
+        bob.expect.greet.called.withArgs({
             a: 1
         });
-        done();
     });
 
     it('allows matching call args with regex', function() {
-        var bob = deride.stub(['greet']);
         bob.greet('The inspiration for this was that my colleague was having a');
         bob.greet({
             a: 123,
@@ -175,7 +174,6 @@ describe('Expectations', function() {
     });
 
     it('allows matching call args with regex', function() {
-        var bob = deride.stub(['greet']);
         bob.greet('The inspiration for this was that my colleague was having a');
 
         (function() {
@@ -184,7 +182,6 @@ describe('Expectations', function() {
     });
 
     it('allows matching call args with regex in objects', function() {
-        var bob = deride.stub(['greet']);
         bob.greet('The inspiration for this was that my colleague was having a');
         bob.greet({
             a: 123,
@@ -195,7 +192,6 @@ describe('Expectations', function() {
     });
 
     it('allows matching call args with regex in deep objects', function() {
-        var bob = deride.stub(['greet']);
         bob.greet('The inspiration for this was that my colleague was having a');
         bob.greet({
             a: 123,

